### PR TITLE
Refactor all of the money.new stuff to use number_to_currency

### PR DESCRIPTION
**Pull request by boveus**
I cleaned up a lot of the tests and all of the references to money.new to use number_to_currency.  This should make it easier to refactor the logic of the cart and the item show pages.  I kept the to_money method on items, since it is in so many places and will need to stick around for the time being.  I don't like the to_money method name, but since the prices are stored in cents in the DB, we will need some method like it to divide the price by 100 before passing it to our views.

I might have missed a few of the prices that weren't set to be a dollar amount. I don't think the tests cover all of the instances of prices being used, so we will need to review the site itself through rails s to catch any I may have missed.

### CODE REVIEW

File: app/models/item.rb [hunk 1]
```diff
@@ -15,7 +15,7 @@ class Item < ApplicationRecord
   enum status: [:active, :retired]
 
   def to_money
-    "$#{Money.new(price, "USD")}"
+    price / 100
```
Comment (JoelLindow) [inline app/models/item.rb]: Love it. Much clearer.

File: spec/features/admin/admin_can_view_individual_orders_spec.rb [hunk 2]
```diff
@@ -16,8 +16,7 @@
       expect(page).to have_content(order.user.name)
       expect(page).to have_content(order.user.address)
       expect(page).to have_content(order.get_quantity(order.items.first.id))
-      expect(page).to have_content(order.get_item_total(order.items.first.id))
-      expect(page).to have_content(order.items.first.to_money)
+      expect(page).to have_content("$0.05")
```
Comment (JoelLindow) [inline spec/features/admin/admin_can_view_individual_orders_spec.rb]: Much more readable. Good refactor, Brandon.

File: spec/features/orders/user_can_view_single_order_page_spec.rb [hunk 3]
```diff
@@ -23,10 +23,10 @@
     expect(current_path).to eq(order_path(order))
     expect(page).to have_content(item.name)
     expect(page).to have_content(item.description)
-    expect(page).to have_content(item.to_money)
+    expect(page).to have_content("$0.05")
```
Comment (JoelLindow) [inline spec/features/orders/user_can_view_single_order_page_spec.rb]: Much more specific. Good call.

File: spec/models/order_spec.rb [hunk 4]
```diff
@@ -56,14 +56,14 @@
...
-    
+
     it '#get_item_total' do
       user = create(:user)
       order = create(:order, user: user)
       item_list = create_list(:item, 5)
       order.items << item_list
 
-      expect(order.get_item_total(order.items.first.id)).to eq('$0.05')
+      expect(order.get_item_total(order.items.first.id)).to eq(0.05)
```
Comment (JoelLindow) [inline spec/models/order_spec.rb]: Way better not in string form.

**Comment by JoelLindow** [comment]
Looks Good, Brandon. Glad you got a jump on this.

File: app/models/item.rb [hunk 5]
```diff
@@ -25,11 +25,16 @@ def set_image
   end
 
   def self.by_popularity
-    select('items.*, count(order_items.id) as order_items_count').joins(:order_items).group(:id).order('order_items_count DESC')
+    select('items.*, count(order_items.id) as order_items_count')
+    .joins(:order_items)
+    .group(:id)
+    .order('order_items_count DESC')
   end
```
Comment (samuelssnider) [inline app/models/item.rb]: I like this, much more legible.

File: app/models/order.rb [hunk 6]
```diff
@@ -29,10 +29,10 @@ def self.get_status_count
...
 
   def item_total_when_ordered(item_id)
-     "$#{Money.new((get_quantity(item_id) * order_items.find_by(item_id: item_id).item_price_record))}"
+    get_quantity(item_id) * order_items.find_by(item_id: item_id).item_price_record / 100
   end
 
   def total_price_when_ordered
-    "$#{Money.new(order_items.sum(:item_price_record))}"
+    order_items.sum(:item_price_record) / 100
   end
```
Comment (samuelssnider) [inline app/models/order.rb]: This looks much cleaner with the refactor!

**Pull request merged**